### ISSUE TITLE
List vs. nested structure for weights, gradients, and updates (dualized grads)

### ISSUE THREAD

**Issue by EIFY** [modula-systems/modula#11]
What's the rationale for using a flat list for weights / gradients / updates instead of a nested structure? I believe the latter is more standard for JAX (see [Working with pytrees](https://docs.jax.dev/en/latest/working-with-pytrees.html)) and avoids the need of counting and slicing like this:

https://github.com/modula-systems/modula/blob/ede2ba72a1b9de3e1f44156db058b5c32c682941/modula/abstract.py#L107-L109
